### ISSUE TITLE
Fix needed for 64bit int builds

### ISSUE THREAD

**Issue by stefanozampini** [mfem/mfem#1283]
I have just noticed this. Probably it got fixed in some branch and will be propagated to master before the release. Would it make sense to add a 64bit builds to the CI in github?

```
/home/szampini/Devel/jointinversion/pkgs/petsc/arch-test-mfem/externalpackages/git.mfem/linalg/hypre.cpp: In member function ‘void mfem::HypreParMatrix::EliminateRows(const mfem::Array<int>&)’:
/home/szampini/Devel/jointinversion/pkgs/petsc/arch-test-mfem/externalpackages/git.mfem/linalg/hypre.cpp:1429:77: error: cannot convert ‘const int*’ to ‘const HYPRE_Int*’ {aka ‘const long long int*’}
 1429 |       internal::hypre_ParCSRMatrixEliminateRows(A, rows.Size(), rows.GetData());
      |                                                                 ~~~~~~~~~~~~^~
      |                                                                             |
      |                                                                             const int*
```

**Comment by jandrej** [comment]
I think this is a good idea. Fixing these sorts of things is on the roadmap also mentioned here #1276.

**Comment by stefanozampini** [comment]
Patch here

```
diff --git a/linalg/hypre.cpp b/linalg/hypre.cpp
index 42c903b7..050d777f 100644
--- a/linalg/hypre.cpp
+++ b/linalg/hypre.cpp
@@ -1426,7 +1426,9 @@ void HypreParMatrix::EliminateRows(const Array<int> &rows)
 {
    if (rows.Size() > 0)
    {
-      internal::hypre_ParCSRMatrixEliminateRows(A, rows.Size(), rows.GetData());
+      Array<HYPRE_Int> r_sorted;
+      get_sorted_rows_cols(rows, r_sorted);
+      internal::hypre_ParCSRMatrixEliminateRows(A, r_sorted.Size(), r_sorted.GetData());
    }
 }
 

```

**Comment by tzanio** [comment]
Fixed in be79c7e08c57bd59675de6721d8d5be43f3b9f05
